src/js/base/runtime.js

```diff
@@ -2282,7 +2282,9 @@ function (Namespace, jsnums, codePoint, util, exnStackParser, loader, seedrandom
     // JS function from Pyret values to Pyret booleans (or throws)
     function equalAlways(v1, v2) {
       if (arguments.length !== 2) { var $a=new Array(arguments.length); for (var $i=0;$i<arguments.length;$i++) { $a[$i]=arguments[$i]; } throw thisRuntime.ffi.throwArityErrorC(["equal-always"], 2, $a, false); }
-      if(typeof v1 === "number" || typeof v1 === "string" || typeof v1 === "boolean") {
+      if (typeof v1 === 'number'  && typeof v2 === 'number' ||
```

Comment:
Stylistically, it bothers me that this test is mixing `&&`, `||` and `===` without any parens -- I think my Pyret is showing :)

Reply:
I can change it, but please note that JS follows precedence conventions commonly followed in every written English text on logic that don't even bother to mention operator precedence, i.e., it is not arbitrary.  Arithmetic binds tighter than conjunction binds tighter than disjunction...

Reply:
Parens added now.